leme-onboarding/src/components/Components/Form.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import styled from 'styled-components';
+
+const Input = styled.input.attrs(props => ({
+    // we can define static props
+    type: props.type,
+  
+    // or we can define dynamic ones
+    size: props.size || "0.1em",
+  }))`
+    color: darkgreen;
+    font-size: 1em;
+    border: 3px solid darkgreen;
+    border-radius: 15px;
+    text-align:center;
+  
+    /* here we use the dynamically computed prop */
+    margin: ${props => props.size};
+    padding: ${props => props.size};
+  `;
+
+interface FormState{
+    type: string;
+    name: string;
+    placeHolder?: string;
+    onChangeFunction: any
+}
+
+interface FormProps {
+    type: string;
+    name: string;
+    placeHolder?: string;
+    onChangeFunction: any
+}
+
+export class Form extends React.Component<FormProps, FormState> {
+
+    constructor (props: any) {
+        super(props)
+
+        this.state = {
+            type: "",
+            name: "",
+            placeHolder: "Insert value",
+            onChangeFunction: null
+        }
+    }
+
+    componentDidMount () {
+        this.setState({
+            type: this.props.type,
+            name: this.props.name,
+            onChangeFunction: this.props.onChangeFunction
+        })
+    }
+
+    render () {
+        return (
+            <div>
+                <Input type={this.state.type} name={this.state.name} placeHolder={this.state.placeHolder} onChange={this.state.onChangeFunction}/>
+            </div>
+        );
+    }
```

Comment:
No caso do form, ele poderia ter mais coisas componentizáveis para que sejam reaproveitáveis no projeto como um todo, o que é justamente o intuito dessa componentização. Todo input vem acompanhado de uma label que indica o que é aquele input e de um caption que pode aparecer quando o campo está inválido. Ou seja, eles poderiam estar no mesmo componente. Além disso, todo form (conjunto label + input + caption) tem o seu estado de erro, que poderia ser definido, por exemplo, pelo aparecimento do caption e por uma borda vermelha no input. Esse tipo de composição de componentes menores e de comportamentos que **devem se repetir** em vários lugares em que for usado, é importante que fiquem centralziados num lugar só. Porque se for preciso alterar algumas dessas propriedades, muito provavelmente um lugar só de alteração no código seria suficiente pra manter esses comportamentos em todos os lugares.

Reply:
Eu estou fazendo a validacão de uma vez só, e não estou recebendo os erros de cada form individualmente. Tem que modificar bastante coisa, vou alterando aqui...

Reply:
Resolvido!